src/pages/Settings/UsersAndPermissions/Groups/RoleSelectField.tsx

```diff
@@ -0,0 +1,135 @@
+import Paper from "@mui/material/Paper";
+import Box from "@mui/material/Box";
+import FormLabel from "@mui/material/FormLabel";
+import Stack from "@mui/material/Stack";
+import { startCase } from "lodash-es";
+import FormControlLabel from "@mui/material/FormControlLabel";
+import Checkbox from "@mui/material/Checkbox";
+import Accordion from "@mui/material/Accordion";
+import AccordionSummary from "@mui/material/AccordionSummary";
+import AccordionDetails from "@mui/material/AccordionDetails";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import Typography from "@mui/material/Typography";
+import { ChangeEvent, useState } from "react";
+import { FieldProps } from "formik";
+
+import { useShop } from "@containers/ShopProvider";
+import { useGetRolesQuery } from "@graphql/generates";
+import { client } from "@graphql/graphql-request-client";
+import { InputWithLabel } from "@components/TextField";
+import { filterNodes } from "@utils/common";
+
+type RoleItem = {
+  name: string
+  label: string
+}
+
+const normalizeRoles = (roleNames: string[]) => {
+  const groupedRoleByResource = roleNames.reduce((roleByResource, roleName) => {
+    const [resource, action] = roleName.split("/");
+    (roleByResource[resource] || (roleByResource[resource] = [])).push({ name: roleName, label: startCase(action) });
+    return roleByResource;
+  }, {} as Record<string, RoleItem[]>);
+
+  return groupedRoleByResource;
+};
+
+const getResourceLabel = (name: string) => startCase(name.split(":")[2] || "Unknown");
+
+type RoleSelectFieldProps = FieldProps<Record<string, string[]>> & {
+  predefinedRoles: string[]
+}
+
+export const RoleSelectField = ({ field: { value: selectedRoles, name: fieldName }, form: { setFieldValue }, predefinedRoles }: RoleSelectFieldProps) => {
+  const { shopId } = useShop();
+
+  const [allRolesByResource, setRolesByResource] =
+  useState(normalizeRoles(predefinedRoles));
+
+
+  const { data } = useGetRolesQuery(client, { shopId: shopId! }, {
+    select: (response) => {
+      const roleNames = [...new Set(filterNodes(response.roles?.nodes).map(({ name }) => name).concat(predefinedRoles))];
```

Comment:
I misunderstood the `getRoles` API, I thought it would return some roles that might not be defined in the default groups. So we need to merge them both to have a complete list. Agree that we can just use the get roles api